lib/Driver/ToolChains/FreeBSD.cpp

```diff
@@ -204,7 +204,7 @@ void freebsd::Linker::ConstructJob(Compilation &C, const JobAction &JA,
 
   // The FreeBSD/MIPS version of GNU ld is horribly buggy and errors out
   // complaining about linking 32-bit and 64-bit code when linking CHERI code.
-  if (IsSandboxABI)
+  if (IsCheriPureCapABI)
     CmdArgs.push_back(Args.MakeArgString("--no-warn-mismatch"));
```

Comment:
I think this might still be needed as long as we build most of our MIPS binaries with GCC